subprojects/test-fixtures/src/main/kotlin/org/gradle/kotlin/dsl/fixtures/SimplifiedKotlinScriptEvaluator.kt

```diff
@@ -56,12 +56,30 @@ import java.io.File
 import java.net.URLClassLoader
 
 
+/**
+ * Evaluates the given Kotlin [script] against this [Project] writing compiled classes
+ * to sub-directories of `$projectDir/.gradle/kotlin-dsl-eval-cache`.
+ */
+fun Project.eval(
+    script: String,
+    baseCacheDir: File = file(".gradle/kotlin-dsl-eval-cache"),
```

Comment:
:x: this creates interference among the tests because:
1. the simplified evaluator only hashes the content of the script
2. it always overwrites the cache contents

I'd move this utility to a base test class where it can set the default baseCacheDir pointing to its own temp folder.